plugins/pipenv/pipenv.plugin.zsh

```diff
@@ -0,0 +1,51 @@
+#compdef pipenv
+_pipenv() {
+  eval $(env COMMANDLINE="${words[1,$CURRENT]}" _PIPENV_COMPLETE=complete-zsh  pipenv)
+}
+
+_togglePipenvShell() {
+  # deactivate shell if Pipfile doesn't exit and not in a sub dir
+  if [[ ! -a "$PWD/Pipfile" ]]; then
+      if [[ "$PIPENV_ACTIVE" == 1 ]]; then
+          if [[ "$PWD" != "$pipfile_dir"* ]]; then
+            exit
```

Comment:
Perhaps `cd -` here, so that the parent shell changes back to where it was before cd-ing into the directory with a Pipfile.  Otherwise, cd-ing out of a pipenv project directory exits the `pipenv shell` but the parent shell is still inside the pipenv project directory.